rust/db_handling/src/identities.rs

```diff
@@ -308,9 +334,10 @@ pub fn cut_path(path: &str) -> Result<(String, String), ErrorSigner> {
 /// Generate new identity (api for create_address())
 /// Function is open to user interface
 pub fn try_create_address (seed_name: &str, seed_phrase: &str, path: &str, network_specs_key: &NetworkSpecsKey, database_name: &str) -> Result<(), ErrorSigner> {
-    match derivation_no_pwd_exists(seed_name, path, network_specs_key, database_name)? {
-        Some((multisigner, address_details)) => return Err(<Signer>::address_generation_common(AddressGenerationCommon::DerivationExists(multisigner, address_details, network_specs_key.to_owned()))),
-        None => {
+    match derivation_check(seed_name, path, network_specs_key, database_name)? {
+        DerivationCheck::BadFormat => return Err(ErrorSigner::AddressGeneration(AddressGeneration::Extra(ExtraAddressGenerationSigner::InvalidDerivation))),
```

Comment:
So yeah earlier in the code you return `Ok` in `derivation_check` just to then return `Err` again here, this back-and-forth does not look necessary. You can propagate one type of error into another